x/stakeibc/keeper/grpc_query_lsm_deposits.go

```diff
@@ -0,0 +1,53 @@
+package keeper
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/Stride-Labs/stride/v8/x/stakeibc/types"
+)
+
+func (k Keeper) LSMDeposits(c context.Context, req *types.QueryLSMDepositsRequest) (*types.QueryLSMDepositsResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid request")
+	}
+
+	var allDeposits []types.LSMTokenDeposit
+	ctx := sdk.UnwrapSDKContext(c)
+
+	// chainId must be included, but denom and status are optional felds
+
+	// Case 1: They are asking for a specific chainId + denom --> GetLSMTokenDeposit
+	// 	 if they included a status and a denom.... ignore the status and search for denom
+	if req.GetDenom() != "" {
+		deposit, found := k.GetLSMTokenDeposit(ctx, req.GetChainId(), req.GetDenom())
+		if found {
+			allDeposits = append(allDeposits, deposit)
+		}
+	}
+
+	// Case 2: They are asking for all deposits with chainId + status --> Get
+	//   if they included a status and denom, already handled above, denom is missing for this branch
+	if req.GetXStatus() != nil && req.GetDenom() == "" {
```

Comment:
is the `X` notation how go + proto handles optionals?

Reply:
Yes, inside the generated pb file from the protos a field which has optional has X versions of each field, these X versions do a bunch of checking around nulls etc.  The reason I have to use the XStatus != nil check is because Status is an enum, the default value of the enum corresponds to the first state (0 int32 under the hood) which is a valid Status and not what we want.  So if they put in nothing for Status by the time we cast it to the actual LSMDepositStatus type it now looks like the first state (in the proto language manual they actually recommend you make the first state of an enum explicitly a catchall value like STATUS_UNDEFINED because of this https://protobuf.dev/programming-guides/proto3/#enum)

Reply:
ah I'm glad you mentioned that. I think we should adapt that as well so the 0-value of our enum's isn't just the first status!